pkgs/by-name/bk/BK7231GUIFlashTool/package.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, mono
+, gtk2
+, imagemagick
+, makeWrapper
+, copyDesktopItems
+, makeDesktopItem
+}:
+
+stdenv.mkDerivation rec {
+  pname = "BK7231GUIFlashTool";
+  version = "50";
+
+  src = fetchFromGitHub {
+    owner = "openshwprojects";
+    repo = "BK7231GUIFlashTool";
+    tag = "v${version}";
+    hash = "sha256-ssL5hX64maAl9qrJ08nUGfEWWLGfWbyv8St5WD4If2Y=";
+  };
+
+  nativeBuildInputs = [
+    mono
+    imagemagick
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    gtk2
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    xbuild BK7231Flasher.sln /p:Configuration=Release
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    mkdir -p $out/lib/BK7231GUIFlashTool
+    cp -r BK7231Flasher/bin/Release/* $out/lib/BK7231GUIFlashTool/
+    makeWrapper ${mono}/bin/mono $out/bin/BK7231Flasher \
+      --add-flags "$out/lib/BK7231GUIFlashTool/BK7231Flasher.exe" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ gtk2 ]}"
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "GUI flashing tool for BK7231 chips used in smart home devices";
+    longDescription = ''
+      BK7231 GUI Flash Tool a simple Windows application that allows you to flash the
+      OpenBK firmware to Beken chipset (BK7231T or BK7231N, can also flash BK7231M,
+      BL2028N and BK7238) based devices without having extensive programming knowledge.
+    '';
+    homepage = "https://github.com/openshwprojects/BK7231GUIFlashTool";
+    license = licenses.mit;
```

Comment:
this would have to be changed, upstream doesn't specify a license yet:
https://github.com/openshwprojects/BK7231GUIFlashTool/issues/62